### DISCUSSION TITLE
Limit and RangeStart queries do not work intuitively

### DISCUSSION THREAD

**Discussion by joshwalsh** [solspace/craft-calendar#104] [Feature Requests]
Hello,

If you create an event that repeats, I expected you could use a query like the following to see the next 10 events:

```
{% set events = craft.calendar.events({ rangeStart: 'today'}).relatedTo(entry).limit(10) %}
```

In fact, this gives you 10 events starting at the date the defined on the calendar event, not 10 from the supplied rangeStart.   If your event repeated weekly with a limit of 10, on week 11 you wouldn't see any new events.  Rather, I expected to see the next 10 repeating events from the supplied rangeStart.

Knowing Craft/Yii inner workings, this makes sense to me as its just adding a LIMIT to the SQL query, but it was a bit unintuitive and caused us some troubleshooting.

I'm not sure if I'd recommend changing the plugin to accommodate this situation or not.  It feels like it could be relatively common.  But, it also would make it act differently than other Craft query builders.  I leave it in your capable hands to decide.

In case anyone else runs into a similar issue, we solved our need by replacing the limit with a rangeEnd instead:

```
{% set events = craft.calendar.events({ rangeStart: 'today', rangeEnd: '3 months' }).relatedTo(entry) %}
```

Josh

**Comment by kjmartens** [comment]
Thanks for your suggestion and details on this one, @joshwalsh. We'll give this one some thought and see if we should make any adjustments. 🙂
